src/keywords/optionalint.cpp

```diff
@@ -55,9 +56,26 @@ std::string OptionalIntegerKeyword::textWhenNull() const { return textWhenNull_;
 // Deserialise from supplied LineParser, starting at given argument offset
 bool OptionalIntegerKeyword::deserialise(LineParser &parser, int startArg, const CoreData &coreData)
 {
+    try
+    {
+        std::stoi(data);
+    }
```

Comment:
So here you can just use the original (actually, fixed!) `parser.argi(startArg)` call to attempt the initial conversion.  This uses `stoi` so will throw an exception if it isn't a valid conversion.

```suggestion
    std::optional<int> newValue;
    try
    {
        newValue = parser.argi(startArg);
    }
```
If this succeeds (i.e. doesn't throw) then your `newValue` optional will contain the value to set.